### PR TITLE
HADOOP-16193. add extra S3A MPU test to see what happens if a file is created during the MPU

### PR DESCRIPTION
**Pull request by steveloughran**
HADOOP-16193. add extra S3A MPU test to see what happens if a file is created during the MPU
    
Change-Id: I15942ff8c7a772e1bf718bccbe4a249d20fa3ef2

### CODE REVIEW

**Comment by steveloughran** [comment]
Tested: S3 ireland. Remember to use -Dscale when running this test suite, even though this case isn't a slow one

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 22 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1004 | trunk passed |
| +1 | compile | 29 | trunk passed |
| +1 | checkstyle | 20 | trunk passed |
| +1 | mvnsite | 34 | trunk passed |
| +1 | shadedclient | 647 | branch has no errors when building and testing our client artifacts. |
| +1 | findbugs | 44 | trunk passed |
| +1 | javadoc | 24 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 28 | the patch passed |
| +1 | compile | 27 | the patch passed |
| +1 | javac | 27 | the patch passed |
| +1 | checkstyle | 16 | the patch passed |
| +1 | mvnsite | 31 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 684 | patch has no errors when building and testing our client artifacts. |
| +1 | findbugs | 46 | the patch passed |
| +1 | javadoc | 19 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 275 | hadoop-aws in the patch passed. |
| +1 | asflicense | 25 | The patch does not generate ASF License warnings. |
| | | 3052 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-609/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/609 |
| Optional Tests |  dupname  asflicense  compile  javac  javadoc  mvninstall  mvnsite  unit  shadedclient  findbugs  checkstyle  |
| uname | Linux a628993d296c 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 091a664 |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| findbugs | v3.1.0-RC1 |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-609/1/testReport/ |
| Max. process+thread count | 446 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-aws U: hadoop-tools/hadoop-aws |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-609/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/contract/s3a/ITestS3AContractMultipartUploader.java [hunk 1]
```diff
@@ -159,4 +170,47 @@ public void testDirectoryInTheWay() throws Exception {
...
+    LOG.info("0-byte file has been created: {}", touchStatus);
+    fs.delete(path, false);
+    // now complete the upload
+    mpu.complete(path, handles, upload1);
+
+    // wait for the data to arrive
+    eventually(timeToBecomeConsistentMillis(), 500, () -> {
+      FileStatus mpuStatus = fs.getFileStatus(path);
+      assertTrue("File is empty in " + mpuStatus, mpuStatus.getLen() > 0);
+      return mpuStatus;
```
Comment (bgaborg) [inline hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/contract/s3a/ITestS3AContractMultipartUploader.java]: Once we've checked the file is there can we delete that file? Or is it not necessary? (Cleanup at the end of the test)

Comment (steveloughran) [reply hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/contract/s3a/ITestS3AContractMultipartUploader.java]: paths created with path() get a unique path that should be cleaned up after; see `AbstractFSContractTestBase.deleteTestDirInTeardown()`

**Comment by steveloughran** [comment]
Aaron has reported a failure, marking as failed

File: hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/contract/s3a/ITestS3AContractMultipartUploader.java [hunk 2]
```diff
@@ -159,4 +170,47 @@ public void testDirectoryInTheWay() throws Exception {
...
+    // confirm the path is absent
+    ContractTestUtils.assertPathDoesNotExist(fs,
+        "path being uploaded", path);
+    // now create an empty file
+    ContractTestUtils.touch(fs, path);
+    final FileStatus touchStatus = fs.getFileStatus(path);
+    LOG.info("0-byte file has been created: {}", touchStatus);
+    fs.delete(path, false);
+    // now complete the upload
+    mpu.complete(path, handles, upload1);
```
Comment (ehiggs) [inline hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/contract/s3a/ITestS3AContractMultipartUploader.java]: Is there anything we can assert about the PathHandle returned here?

Comment (steveloughran) [reply hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/contract/s3a/ITestS3AContractMultipartUploader.java]: that it's not empty?

Comment (ehiggs) [reply hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/contract/s3a/ITestS3AContractMultipartUploader.java]: That you can use it to open the file again and read it?

Comment (steveloughran) [reply hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/contract/s3a/ITestS3AContractMultipartUploader.java]: You may be able to use a PathHandle but these aren't, are they? They're PathHandles: Serializable representations of an uploaded file part.

Comment (ehiggs) [reply hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/contract/s3a/ITestS3AContractMultipartUploader.java]: No, I mean mpu.complete returns a **Path**Handle. The part handles are **Part**Handles. Indeed PartHandles aren't useful out of the context of putPart (intentionally). But PathHandle is indeed useful. :)

Comment (steveloughran) [reply hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/contract/s3a/ITestS3AContractMultipartUploader.java]: I don't think I fully understand pathhandles. I thought that was an HDFS thing only right, or is there something obvious I haven't noticed?

Pauses, looks at code. I see: S3A MPU returns the etag, which isn't enough to refer to it when reopening. 

We've been doing work on S3 & versioning now: one thing I'd like to get back from the upload is a reference which can be guaranteed to open that specific file, or failfast if it has changed; something like (path, etag, version). Returning these immediately from the upload in an S3AFileStatus would be one approach, but a pathhandle for S3A which contained all of those would be exactly what someone needs, isn't it?

Do you fancy getting busy on that?

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 57 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1125 | trunk passed |
| +1 | compile | 33 | trunk passed |
| +1 | checkstyle | 19 | trunk passed |
| +1 | mvnsite | 42 | trunk passed |
| +1 | shadedclient | 784 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 21 | trunk passed |
| 0 | spotbugs | 55 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 54 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 35 | the patch passed |
| +1 | compile | 32 | the patch passed |
| +1 | javac | 32 | the patch passed |
| +1 | checkstyle | 15 | the patch passed |
| +1 | mvnsite | 32 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 713 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 21 | the patch passed |
| +1 | findbugs | 66 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 277 | hadoop-aws in the patch passed. |
| +1 | asflicense | 25 | The patch does not generate ASF License warnings. |
| | | 3426 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.8 Server=18.09.8 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-609/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/609 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 0b7b25039e89 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / cd967c7 |
| Default Java | 1.8.0_212 |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-609/2/testReport/ |
| Max. process+thread count | 411 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-aws U: hadoop-tools/hadoop-aws |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-609/2/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 81 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1305 | trunk passed |
| +1 | compile | 40 | trunk passed |
| +1 | checkstyle | 23 | trunk passed |
| +1 | mvnsite | 46 | trunk passed |
| +1 | shadedclient | 822 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 25 | trunk passed |
| 0 | spotbugs | 70 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| -1 | findbugs | 67 | hadoop-tools/hadoop-aws in trunk has 1 extant findbugs warnings. |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 39 | the patch passed |
| +1 | compile | 37 | the patch passed |
| +1 | javac | 37 | the patch passed |
| +1 | checkstyle | 21 | the patch passed |
| +1 | mvnsite | 45 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 901 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 27 | the patch passed |
| +1 | findbugs | 74 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 294 | hadoop-aws in the patch passed. |
| +1 | asflicense | 28 | The patch does not generate ASF License warnings. |
| | | 3951 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-609/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/609 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 2b6db0ae4f72 4.15.0-52-generic #56-Ubuntu SMP Tue Jun 4 22:49:08 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / c7c7a88 |
| Default Java | 1.8.0_212 |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-609/3/artifact/out/branch-findbugs-hadoop-tools_hadoop-aws-warnings.html |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-609/3/testReport/ |
| Max. process+thread count | 349 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-aws U: hadoop-tools/hadoop-aws |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-609/3/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 168 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1411 | trunk passed |
| +1 | compile | 38 | trunk passed |
| +1 | checkstyle | 30 | trunk passed |
| +1 | mvnsite | 55 | trunk passed |
| +1 | shadedclient | 902 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 28 | trunk passed |
| 0 | spotbugs | 64 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| -1 | findbugs | 62 | hadoop-tools/hadoop-aws in trunk has 1 extant findbugs warnings. |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 35 | the patch passed |
| +1 | compile | 36 | the patch passed |
| +1 | javac | 36 | the patch passed |
| +1 | checkstyle | 21 | the patch passed |
| +1 | mvnsite | 42 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 934 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 29 | the patch passed |
| +1 | findbugs | 74 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 295 | hadoop-aws in the patch passed. |
| +1 | asflicense | 36 | The patch does not generate ASF License warnings. |
| | | 4268 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-609/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/609 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux ec74f0e23d67 4.15.0-48-generic #51-Ubuntu SMP Wed Apr 3 08:28:49 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / e20b195 |
| Default Java | 1.8.0_212 |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-609/4/artifact/out/branch-findbugs-hadoop-tools_hadoop-aws-warnings.html |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-609/4/testReport/ |
| Max. process+thread count | 323 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-aws U: hadoop-tools/hadoop-aws |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-609/4/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 103 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1257 | trunk passed |
| +1 | compile | 31 | trunk passed |
| +1 | checkstyle | 21 | trunk passed |
| +1 | mvnsite | 35 | trunk passed |
| +1 | shadedclient | 775 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 24 | trunk passed |
| 0 | spotbugs | 55 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 54 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 31 | the patch passed |
| +1 | compile | 26 | the patch passed |
| +1 | javac | 26 | the patch passed |
| +1 | checkstyle | 17 | the patch passed |
| +1 | mvnsite | 30 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 805 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 22 | the patch passed |
| +1 | findbugs | 59 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 281 | hadoop-aws in the patch passed. |
| +1 | asflicense | 30 | The patch does not generate ASF License warnings. |
| | | 3693 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-609/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/609 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 57d7b489d4ad 4.15.0-48-generic #51-Ubuntu SMP Wed Apr 3 08:28:49 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / b0131bc |
| Default Java | 1.8.0_212 |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-609/5/testReport/ |
| Max. process+thread count | 340 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-aws U: hadoop-tools/hadoop-aws |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-609/5/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 91 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1235 | trunk passed |
| +1 | compile | 37 | trunk passed |
| +1 | checkstyle | 23 | trunk passed |
| +1 | mvnsite | 37 | trunk passed |
| +1 | shadedclient | 772 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 24 | trunk passed |
| 0 | spotbugs | 58 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 55 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 32 | the patch passed |
| +1 | compile | 26 | the patch passed |
| +1 | javac | 26 | the patch passed |
| +1 | checkstyle | 17 | the patch passed |
| +1 | mvnsite | 30 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 848 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 27 | the patch passed |
| +1 | findbugs | 76 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 93 | hadoop-aws in the patch passed. |
| +1 | asflicense | 30 | The patch does not generate ASF License warnings. |
| | | 3551 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-609/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/609 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 5608c1a354dd 4.15.0-54-generic #58-Ubuntu SMP Mon Jun 24 10:55:24 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / e356e4f |
| Default Java | 1.8.0_222 |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-609/6/testReport/ |
| Max. process+thread count | 306 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-aws U: hadoop-tools/hadoop-aws |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-609/6/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 107 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1214 | trunk passed |
| +1 | compile | 30 | trunk passed |
| +1 | checkstyle | 22 | trunk passed |
| +1 | mvnsite | 36 | trunk passed |
| +1 | shadedclient | 812 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 24 | trunk passed |
| 0 | spotbugs | 58 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 56 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 32 | the patch passed |
| +1 | compile | 26 | the patch passed |
| +1 | javac | 26 | the patch passed |
| +1 | checkstyle | 19 | the patch passed |
| +1 | mvnsite | 31 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 833 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 22 | the patch passed |
| +1 | findbugs | 61 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 70 | hadoop-aws in the patch passed. |
| +1 | asflicense | 28 | The patch does not generate ASF License warnings. |
| | | 3508 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-609/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/609 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux a74ea948554c 4.15.0-54-generic #58-Ubuntu SMP Mon Jun 24 10:55:24 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 094d736 |
| Default Java | 1.8.0_222 |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-609/7/testReport/ |
| Max. process+thread count | 342 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-aws U: hadoop-tools/hadoop-aws |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-609/7/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/contract/s3a/ITestS3AContractMultipartUploader.java (hunk 2 above)
Comment (ehiggs) [reply hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/contract/s3a/ITestS3AContractMultipartUploader.java]: >one thing I'd like to get back from the upload is a reference which can be guaranteed to open that specific file

That does sound better.  We should make a ticket as soon as apache jira is back online.

**Comment by ehiggs** [comment]
LGTM!

**Comment by ehiggs** [comment]
Merged.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 0 | Docker mode activated. |
| -1 | patch | 13 | https://github.com/apache/hadoop/pull/609 does not apply to trunk. Rebase required? Wrong Branch? See https://wiki.apache.org/hadoop/HowToContribute for help. |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/609 |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-609/8/console |
| versions | git=2.7.4 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 49 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1273 | trunk passed |
| +1 | compile | 36 | trunk passed |
| +1 | checkstyle | 24 | trunk passed |
| +1 | mvnsite | 42 | trunk passed |
| +1 | shadedclient | 875 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 28 | trunk passed |
| 0 | spotbugs | 71 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 69 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 35 | the patch passed |
| +1 | compile | 30 | the patch passed |
| +1 | javac | 30 | the patch passed |
| +1 | checkstyle | 21 | the patch passed |
| +1 | mvnsite | 34 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 901 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 26 | the patch passed |
| +1 | findbugs | 70 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 86 | hadoop-aws in the patch passed. |
| +1 | asflicense | 32 | The patch does not generate ASF License warnings. |
| | | 3722 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-609/9/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/609 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux f9dc8f84311c 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 3329257 |
| Default Java | 1.8.0_222 |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-609/9/testReport/ |
| Max. process+thread count | 411 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-aws U: hadoop-tools/hadoop-aws |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-609/9/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 42 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1102 | trunk passed |
| +1 | compile | 34 | trunk passed |
| +1 | checkstyle | 24 | trunk passed |
| +1 | mvnsite | 36 | trunk passed |
| +1 | shadedclient | 728 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 27 | trunk passed |
| 0 | spotbugs | 62 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 60 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 33 | the patch passed |
| +1 | compile | 32 | the patch passed |
| +1 | javac | 32 | the patch passed |
| +1 | checkstyle | 21 | the patch passed |
| +1 | mvnsite | 33 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 744 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 28 | the patch passed |
| +1 | findbugs | 66 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 84 | hadoop-aws in the patch passed. |
| +1 | asflicense | 29 | The patch does not generate ASF License warnings. |
| | | 3210 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-609/10/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/609 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 43ade3e885eb 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 915cbc9 |
| Default Java | 1.8.0_222 |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-609/10/testReport/ |
| Max. process+thread count | 447 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-aws U: hadoop-tools/hadoop-aws |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-609/10/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by steveloughran** [comment]
ok, if this doesn't work as is -what should our next step be?

**Pull request closed without merging**